### PR TITLE
Associate demo files with actionfps.

### LINKED ISSUE #133: associate .dmo files to actionfps on Linux
## What?
Opening a .dmo file should launch the game and open the demo.

## Why?
 * Loading a demo from the game is quite confusing. The GUI requires it to be in a specific location and when there are two many of them they are difficult to find.
 * Loading from the command line is not user friendly
 * Should make debugging easier for us.

## How:
File associaton:
 * Windows: http://nsis.sourceforge.net/File_Association
 * Linux: register mime type with xdg-mime
 * Mac: no idea
 * parameter to pass: --loaddemo=path

### PR DESCRIPTION
**Pull request by lucasgautheron**
## Original issue
Closes #133.

## Description
Associate demo files with actionfps on all platforms.

## Check List

* [x] Linux file association
* [x] Demos can't be read from any location
* [x] Extension changed to .dmo.gz
* [x] Locally tested

### CODE REVIEW

File: source/src/stream.cpp [hunk 1]
```diff
@@ -886,9 +893,9 @@ stream *openrawfile(const char *filename, const char *mode)
     return file;
 }
 
-stream *openfile(const char *filename, const char *mode)
+stream *openfile(const char *filename, const char *mode, bool allow_outside_game)
 {
-    const char *found = findfile(filename, mode);
+    const char *found = findfile(filename, mode, allow_outside_game);
 #ifndef STANDALONE
     if(!strncmp(found, "zip://", 6)) return openzipfile(found + 6, mode);
```
Comment (ScalaWilliam) [inline source/src/stream.cpp]: who zips a gzipped file 🤕

**Comment by ScalaWilliam** [comment]
This can be merged by itself, extension association can be a separate thing?

**Comment by lucasgautheron** [comment]
Yes, we can even merge without closing this PR

File: source/src/stream.cpp (hunk 1 above)
Comment (lucasgautheron) [inline source/src/stream.cpp]: openfile is generic :(

**Comment by ScalaWilliam** [comment]
Maybe reducing scope will be useful? Reduce to only Linux; then create Issue for Windows & 1 issue for Mac. I'd like contribute but the changeset will be separate.

**Comment by lucasgautheron** [comment]
Alright.

**Pull request merged**